File: day4/a.ts
const input = await Deno.readTextFile("input.txt");
const pairs = input.trim().split(/\r?\n/);

function main() {
  let count = 0;
  pairs.map((s) => {
    const sections = s.split(/,|-/);
    const [a, b, x, y] = sections.map(Number);
    if (!(b < x || y < a)) {
      count++;
    }
  });
  console.log("Count: ", count);
}

main();

File: day4/a.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	file, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	sc := bufio.NewScanner(file)
	fullyContainedCount := 0
	for sc.Scan() {
		var a, b, x, y int
		fmt.Sscanf(sc.Text(), "%d-%d,%d-%d", &a, &b, &x, &y)
		if (a <= x && b >= y) || (x <= a && y >= b) {
			fullyContainedCount++
		}
	}
	fmt.Printf("count: %d\n", fullyContainedCount)
}
